msm/src/lib.rs

```diff
@@ -29,6 +29,11 @@ pub type BN254 = ark_ec::bn::Bn<ark_bn254::Parameters>;
 pub type BN254G1Affine = <BN254 as ark_ec::PairingEngine>::G1Affine;
 pub type BN254G2Affine = <BN254 as ark_ec::PairingEngine>::G2Affine;
 
+/// Number of columns
+/// FIXME: we must move it into the subdirectory of the
+/// foreign field addition circuit
+pub const N: usize = 3 * LIMBS_NUM;
```

Comment:
`MSM_COLUMN_N`? Can we at least keep single letters for type parameters? There are two `N` in this file...